remotes/errors/errors.go

```diff
@@ -30,23 +35,38 @@ type ErrUnexpectedStatus struct {
 	StatusCode                int
 	Body                      []byte
 	RequestURL, RequestMethod string
+	Message                   string
 }
 
 func (e ErrUnexpectedStatus) Error() string {
-	return fmt.Sprintf("unexpected status from %s request to %s: %s", e.RequestMethod, e.RequestURL, e.Status)
+	return fmt.Sprintf("unexpected status(%s) from %s request to %s: %s", e.Status, e.RequestMethod, e.RequestURL, e.Message)
 }
 
 // NewUnexpectedStatusErr creates an ErrUnexpectedStatus from HTTP response
 func NewUnexpectedStatusErr(resp *http.Response) error {
 	var b []byte
+	var message string
 	if resp.Body != nil {
 		b, _ = io.ReadAll(io.LimitReader(resp.Body, 64000)) // 64KB
+		if len(b) > 0 {
+			var r dtspecv1.ErrorResponse
+			err := json.Unmarshal(b, &r)
+			if err != nil {
+				log.G(context.Background()).Debugf("Unmarshal response body failed: %v", err)
+			}
+			for _, e := range r.Errors {
+				if len(e.Message) > 0 {
+					message = e.Message
+				}
+			}
```

Comment:
Why not all errors are preserved?

Reply:
Will keep all of the messages, most cases the array length is 1. Please review again.